snapshot/snapshot.go

```diff
@@ -275,14 +275,14 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 		if ok, ref, layerDigest, blob := o.fs.SupportStargz(ctx, base.Labels); ok {
 			err = o.fs.PrepareStargzMetaLayer(ctx, blob, ref, layerDigest, s, base.Labels)
 			if err != nil {
-				logCtx.Errorf("failed to prepare stargz layer of snapshot ID %s, err: %v", s.ID, err)
-			} else {
-				// Mark this snapshot as stargz layer
-				base.Labels[label.StargzLayer] = "true"
-				err := o.Commit(ctx, target, key, append(opts, snapshots.WithLabels(base.Labels))...)
-				if err == nil || errdefs.IsAlreadyExists(err) {
-					return nil, errors.Wrapf(errdefs.ErrAlreadyExists, "target snapshot %q", target)
-				}
+				logCtx.Errorf("prepare stargz layer of snapshot ID %s, err: %v", s.ID, err)
+				return nil, errors.Wrap(err, "prepare stargz image layer")
```

Comment:
If prepare stargz meta layer fails, we need to fall back to OCI non-lazy mode, so should ignore the error here.

Reply:
Ok, didn't realized that and will revert it.